src/shared/dissect-image.c

```diff
@@ -841,84 +845,110 @@ static int dissect_image(
 
         uint64_t diskseq = m->loop ? m->loop->diskseq : 0;
 
-        if (verity && iovec_is_set(&verity->root_hash)) {
-                sd_id128_t fsuuid, vuuid;
+        /* If input is a single already open verity device and the filesystem type is pinned to a single
+         * fstype, and it's a single-image filesystem, then skip all the expensive probing and just shortcut
+         * to mount the dm device directly */
+        if ((flags & DISSECT_IMAGE_NO_PARTITION_TABLE)) {
+                PartitionPolicyFlags policy_flags = image_policy_get_exhaustively(policy, PARTITION_ROOT);
+                if (policy_flags < 0)
+                        return policy_flags;
+
+                /* The policy fstype flags translate to the literal fstype name of each filesystem. */
+                r = partition_policy_flags_to_string(policy_flags & _PARTITION_POLICY_FSTYPE_MASK, /* simplify= */ true, &root_fstype_string);
```

Comment:
i'd *really* prefer if this kind of logic is also applied to the case where we do have a partition table. if the policy says /home/ must be btrfs, then we should not test for anything else.

i know this doesn't matter to your usecase, but i think we should follow the same logic comprehensively.

Reply:
hence, maybe turn these lines into a helper:

partition_policy_determine_fstype() or so, which you just hand a policy and a designator, and it returns NULL if the policy doesn't fixate things, and otherwise returns the fstype. then use this here, *and* in the partition codepaths below.

Reply:
In that case though blkid cannot be skipped as it's needed for all the rest - but for the sole purpose of the fstype choice yeah I can add that there too

Reply:
Sure, makes sense